FO-DICOM.Core/DicomDatasetExtensions.cs

```
@@ -71,12 +71,19 @@ public static DateTimeOffset GetDateTimeOffset(this DicomDataset dataset, DicomT
             var timezone = (topLevelDataset ?? dataset).GetDicomItem<DicomShortString>(DicomTag.TimezoneOffsetFromUTC);
             if (timezone != null && timezone.Count > 0)
             {
-                // Explicit timezone information present in dataset
+                // Explicit timezone information present in dataset. The format is &XXZZ, where & is either '+' or '-' and XX or YY are the hours or minutes
                 string s = timezone.Get<string>();
-                int hh = int.Parse(s.Substring(0, 3));
+                DicomValidation.ValidateTimezoneOffset(s);
+                int sign = s[0] switch
+                {
+                    '+' => +1,
+                    '-' => -1,
+                    _ => throw new DicomValidationException(s, DicomVR.SH, "Invalid format for TimezoneOffsetFromUTC")
```

If validation is on, this is already done in `ValidateTimezoneOffset` now. I wonder if the timezone shall be ignored with a warning in case validation is off and the offset is invalid? Not sure about this, though.

---

not exactly. With the paradigma of being tollerant when receiving but strict when creating data, the validation currently _only_ applies if the user AddOrUpdates a new value to TimezoneOffset. If reading or receiving a file, then there is never any validation.
So if a user now calls `GetDatetimeOffset`, then I wanted to make it similar as if someone called `Get<int>` of a value that is a string. If the content is invalid then an exception is thrown insted of silently ignoring.
But I see, a more consistent way would then, to also add a method `TryGetDatetimeOffset` which then never throws and ignores any invalid data.

---

And as written in the comment, the current version did not do any validation, but directly called `int.Parse(s.Substring(3,2))` etc. So if there was an invalid input, then this also would have thrown, but it had thrown some IndexOutOfRangeException or some FormatException or some ArgumentNullException - all of them things that would not give the user a hint what exactly had happened. Now with this check in advance, these kind of exceptions are prevented and instead there is a validationexception that shows the content and tells about the wrong formatting of the DatetimeOffset.

---

>  If reading or receiving a file, then there is never any validation

Ah right, I had forgotten about this.

> But I see, a more consistent way would then, to also add a method TryGetDatetimeOffset which then never throws and ignores any invalid data.

Agreed.

> the current version did not do any validation, but directly called int.Parse(s.Substring(3,2))

Sure, this was clearly a bug, and the new behavior is definitely an improvement. I was just wondering if we would also need a possibility to ignore this kind of invalid data.

---

> Sure, this was clearly a bug, and the new behavior is definitely an improvement. I was just wondering if we would also need a possibility to ignore this kind of invalid data.

Thats a good idea. So I will add an TryGetDatetimeOffset method, that will ignore errors and makes this clear to the user that this method will ignore any invalid input.